### ISSUE TITLE
`selector-class-pattern` pattern does not allow for BEM selector names

### ISSUE THREAD

**Issue by mockdeep** [stylelint/stylelint-config-standard#209]
### What steps are needed to reproduce the bug?

Use classes such as these from [the AirBnB style guide](https://github.com/airbnb/css#oocss-and-bem):

```scss
.ListingCard { }
.ListingCard--featured { }
.ListingCard__title { }
.ListingCard__content { }
```

### What did you expect to happen?

These selectors should not report any issues with the `selector-class-pattern` rule.

### What actually happened?

Output:

```
Expected class selector to be kebab-case  selector-class-pattern
```

It appears [the pattern](https://github.com/stylelint/stylelint-config-standard/blob/6ed92800d32768531aacfe940e95d6a916d47c87/index.js#L125) only allows for single dashes as separators.

**Comment by jeddy3** [comment]
@mockdeep Thanks for the report.

The config is designed to help people write CSS that looks like something you'd find in the examples of the specifications.

For more advanced use cases, e.g. an involved naming methodology like BEM, you'll need to extend and customise this config.

You should turn off the `selector-class-pattern` rule (using `null`) and I suggest using the [stylelint-selector-bem-pattern](https://www.npmjs.com/package/stylelint-selector-bem-pattern) plugin instead. It's very powerful and very good at enforcing BEM conventions.
